src/pybind/mgr/cephadm/module.py

```diff
@@ -2172,16 +2202,16 @@ def _create_osd(self, host, cmd):
                 if osd['tags']['ceph.cluster_fsid'] != fsid:
                     self.log.debug('mismatched fsid, skipping %s' % osd)
                     continue
-                if osd_id in before_osd_uuid_map:
-                    # this osd existed before we ran prepare
+                if osd_id in before_osd_uuid_map and osd_id not in replace_osd_ids:
+                    # if it exists but is part of the replacement operation, don't skip
                     continue
                 if osd_id not in osd_uuid_map:
                     self.log.debug('osd id %d does not exist in cluster' % osd_id)
                     continue
-                if osd_uuid_map[osd_id] != osd['tags']['ceph.osd_fsid']:
+                if osd_uuid_map.get(osd_id) != osd['tags']['ceph.osd_fsid']:
```

Comment:
`osd_uuid_map.get(osd_id)` is basically the same as `osd_uuid_map.get[osd_id]`. Maybe

```suggestion
                if osd_uuid_map.get(osd_id, None) != osd['tags']['ceph.osd_fsid']:
```
?

Reply:
It's not quite the same:

while

```
In [3]: {}['foo']                                                               
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-3-e91fc3e9cce0> in <module>
----> 1 {}['foo']

KeyError: 'foo'
```

raises a KeyError,

the .get() method for dictionaries returns `None` when the key was not found. 

``` python
In [2]: {}.get('foo') is None                                                   
Out[2]: True
```

Hence, 

```python
{}.get('foo', None)
```

is redundant